### DISCUSSION TITLE
Does FakeXRMEasy supports uploading files to file type columns?

### DISCUSSION THREAD

**Discussion by kareru199x** [jordimontana82/fake-xrm-easy#603] [Q&A]
Hi All,

I am currently trying to populate a file column of my entity for my unit testing. I followed the [code](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/file-column-data?tabs=sdk) for Microsoft Learn on how to upload a physical file on the column. But I am getting this error message:
"`_Exception: The organization request type 'Microsoft.Crm.Sdk.Messages.InitializeFileBlocksUploadRequest' is not yet supported... but we DO love pull requests so please feel free to submit one! :). This functionality is not available yet. Please consider contributing to the following Git project https://github.com/jordimontana82/fake-xrm-easy by cloning the repository and issuing a pull request._`"

The service client that I use was this: `fakeContext.GetOrganizationService();
`
Am I doing it wrong? Or is there an alternative way to achieve this?

Thank you.

**Comment by jordimontana82** [comment]
Hi @kareru199x ,

Thanks a mill for your question.

File and Image types are not supported in v1.x.  I've added these to the backlog of the newer, more recent versions (2.x and 3.x): https://github.com/DynamicsValue/fake-xrm-easy/issues/84

Moreover, v1.x is deprecated and no longer actively maintained, you might want to check our migration guide to the most recent version in 6 steps:

https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/migrating-from-1x/

With that said, you can add your own fake message executor with a minimal implementation as a workaround. You can also achieve these in  the newer versions while the file and image support is not yet there. 

Here's an excellent resource by @bwmodular  that covers doing this in detail:

http://www.bwmodular.org/blog/mocking-unimplemented-organisation-requests-in-fakexrmeasy
